src/projections/aeqd.cpp

```diff
@@ -96,7 +96,7 @@ static PJ_XY aeqd_e_forward (PJ_LP lp, PJ *P) {          /* Ellipsoidal, forward
     struct pj_opaque *Q = static_cast<struct pj_opaque*>(P->opaque);
     double  coslam, cosphi, sinphi, rho;
     double azi1, azi2, s12;
-    double lam1, phi1, lam2, phi2;
+    double lat1, lon1, lat2, lon2;
```

Comment:
unrelated to your changes, but looking at that code, I see that coslam, cosphi and sinphi are not used in the EQUIT and OBLIQ cases. I was wondering if the compiler is smart enough to realize that, and in fact it is only 50% smart. It doesn't compute coslam, but probably due to a heuristics when seeing cos(lp.phi) and sin(lp.phi), it calls the sincos() function in all code paths. cf this disassembly of the method where one sees there's no conditional jump before the call to sincos()

```
0000000000118fb0 <aeqd_e_forward(PJ_LP, PJconsts*)>:
  118fb0:       f3 0f 1e fa             endbr64 
  118fb4:       41 55                   push   %r13
  118fb6:       41 54                   push   %r12
  118fb8:       55                      push   %rbp
  118fb9:       53                      push   %rbx
  118fba:       48 89 fb                mov    %rdi,%rbx
  118fbd:       48 83 ec 58             sub    $0x58,%rsp
  118fc1:       4c 8d 64 24 28          lea    0x28(%rsp),%r12
  118fc6:       4c 8d 6c 24 20          lea    0x20(%rsp),%r13
  118fcb:       f2 0f 11 0c 24          movsd  %xmm1,(%rsp)
  118fd0:       64 48 8b 04 25 28 00 00 00      mov    %fs:0x28,%rax
  118fd9:       48 89 44 24 48          mov    %rax,0x48(%rsp)
  118fde:       31 c0                   xor    %eax,%eax
  118fe0:       4c 89 ee                mov    %r13,%rsi
  118fe3:       4c 89 e7                mov    %r12,%rdi
  118fe6:       f2 0f 11 44 24 08       movsd  %xmm0,0x8(%rsp)
  118fec:       e8 ef 4a f7 ff          callq  8dae0 <sincos@plt>
```

This is true of gcc 9.4 in -O2 mode, and clang 10.0 in -O2 (clang is actually worse calls cos(), cos() and sin() !)

Reply:
cf https://github.com/OSGeo/PROJ/pull/3523 for a cleanup done in the spherical forward case which suffers from the same issue (didn't touch the ellipsoidal case to avoid conflicts with this PR)

Reply:
I moved the assignments for cosphi+sinphi to the `S_POLE` case which should take care of this.

If it were up to me (and you and I probably differ here), I would remove the special treatment for `{S,N}_POLE` and let the general code handle all cases.  Pluses: shorter code, easier to maintain, the results in the special cases serve as a check for the general case.  Minus: slower is what might be relatively common special cases.